research-software-directory/RSD-as-a-service
# chore: remove redundant category types and use MUI component for sele…

## Consolidate category types and use MUI

Changes proposed in this pull request:

* Remove the redundant types `CategoryTreeLevel` and `CategoryTree` and use `TreeNode<CategoryEntry>` instead
* Use MUI components for the category selector on the software edit page

How to test:

* `docker compose down --volumes && docker compose build --parallel && docker compose up --scale data-generation=1`
* Sign in as admin
* Create some highlighted categories
* Edit software pages to add and remove categories to them, there should be no errors and the visual state should update immediately
* Check the software view page to see if it's still rendered correctly (nothing should be visually changed here)


Closes #1236

PR Checklist:
